core/src/apps/cardano/sign_tx.py

```diff
@@ -67,522 +84,355 @@
     show_warning_tx_pointer_address,
     show_warning_tx_staking_key_hash,
 )
-from .seed import is_byron_path, is_shelley_path
 
 if False:
-    from typing import Any, Optional, Union
-
-    from trezor.messages.CardanoSignTx import CardanoSignTx
-    from trezor.messages.CardanoTxCertificateType import CardanoTxCertificateType
-    from trezor.messages.CardanoTxInputType import CardanoTxInputType
-    from trezor.messages.CardanoTxOutputType import CardanoTxOutputType
-    from trezor.messages.CardanoTxWithdrawalType import CardanoTxWithdrawalType
-    from trezor.messages.CardanoAssetGroupType import CardanoAssetGroupType
-
-    from apps.common.cbor import CborSequence
+    from typing import Optional
     from apps.common.paths import PathSchema
 
-    CborizedTokenBundle = dict[bytes, dict[bytes, int]]
-    CborizedTxOutput = tuple[bytes, Union[int, tuple[int, CborizedTokenBundle]]]
-    CborizedSignedTx = tuple[dict, dict, Optional[cbor.Raw]]
-    TxHash = bytes
 
 MINTING_POLICY_ID_LENGTH = 28
 MAX_ASSET_NAME_LENGTH = 32
-MAX_TX_CHUNK_SIZE = 256
 
 
 @seed.with_keychain
 async def sign_tx(
     ctx: wire.Context, msg: CardanoSignTx, keychain: seed.Keychain
-) -> CardanoSignedTx:
-    if msg.fee > LOVELACE_MAX_SUPPLY:
-        raise wire.ProcessError("Fee is out of range!")
+) -> CardanoSignTxFinished:
+    is_network_id_verifiable = await _validate_tx_signing_request(ctx, msg)
 
-    validate_network_info(msg.network_id, msg.protocol_magic)
+    builder = TxBuilder(keychain, msg)
+
+    common_input_account = await _process_inputs(ctx, msg, keychain, builder)
+    total_amount_to_show = await _process_outputs(
+        ctx, msg, keychain, builder, common_input_account
+    )
+    builder.add_fee()
+    builder.add_ttl()
+    await _process_certificates(ctx, msg, keychain, builder)
+    await _process_withdrawals(ctx, msg, builder)
+    catalyst_signature = await _process_auxiliary_data(ctx, msg, keychain, builder)
```

Comment:
I find it a bit weird for a function with a generic name like `_process_auxiliary_data()` to return the catalyst signature directly which is specific for a certain kind of auxiliary data. I'd consider returning the AuxiliaryDataSupplement or None type directly and the logic below which returns it could also look more intuitive then and not check for some "catalyst signature" out of the blue